src/uct/ib/rc/base/rc_ep.c

```diff
@@ -642,15 +642,24 @@ uct_rc_ep_check(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
     return UCS_OK;
 }
 
-int uct_rc_ep_is_connected(struct ibv_ah_attr *ah_attr,
-                           const uct_ep_is_connected_params_t *params,
-                           uint32_t qp_num, uint32_t addr_qp)
+static int uct_rc_ep_is_connected(uct_rc_ep_t *ep)
+{
+    return ucs_test_flags(ep->flags, UCT_RC_EP_FLAG_CONNECTED);
+}
+
+int uct_rc_ep_is_connected_to_addr(uct_rc_ep_t *ep, struct ibv_ah_attr *ah_attr,
+                                   const uct_ep_is_connected_params_t *params,
+                                   uint32_t qp_num, uint32_t addr_qp)
 {
     union ibv_gid *rgid;
     const uct_ib_address_t *ib_addr;
 
     UCT_EP_IS_CONNECTED_CHECK_DEV_ADDR(params);
 
+    if (!uct_rc_ep_is_connected(ep)) {
```

Comment:
```
if (!(ep->flags & UCT_RC_EP_FLAG_CONNECTED)) {
    return 0;
}
```

or
```
static int uct_rc_ep_is_connected(uct_rc_ep_t *ep)
{
     return (ep->flags & UCT_RC_EP_FLAG_CONNECTED);
}
```

and change all places that use UCT_RC_EP_FLAG_CONNECTED to use uct_rc_ep_is_connected